### PR TITLE
Force default MSVC linker when building on Windows

### PR DESCRIPTION
**Pull request by IceSentry**
Release Notes:

- N/A

## Description

Using `rust-lld.exe` globally on windows is pretty common since it can help speed up compilation. Unfortunately, zed fails to build with that linker. This PR forces the default "link.exe" to be used on windows to avoid this potential error. This is intended as a temporary fix until the reason why it fails to compile with another linker is found.

Changing the linker from the global setting might seem a bit wrong but the alternative right now is just crashing with a `STATUS_ACCESS_VIOLATION` without any explanations as to what went wrong.

See #12041

### CODE REVIEW

**Comment by IceSentry** [comment]
@cla-bot check

**Comment by osiewicz** [comment]
Hey, I don't think we should merge this change.
1. `rust-lld` is still in experimental phase, so I think it's expected for it to crash on some inputs. I don't think it makes sense to dance around problems with the tooling on dev machine in Zed repository itself, especially when said tooling is still in beta.
2. This change actually prevents users from using their own linker if they wish to do so (e.g. llvm-lld), even when these linkers don't cause an AV for Zed.

Instead, I'd suggest overriding your global default for `rust-lld` by having a parent directory over zed repo that contains a separate `.cargo/config.toml` (and that file would contain the changes from this PR, in a nutshell).
That way, we don't need to maintain that in the repo itself.
So, in short, instead of having zed in `E:\projects\zed`, you'd need to nest it twice like so:
```
E:\projects\zed\zed <- path to the root of this repository
```
There'd also be a sibling directory in `E:\projects\zed`:
`E:\projects\zed\.cargo` that would contain `config.toml` with the overrides you need.

**Comment by IceSentry** [comment]
Right, I didn't consider a nested folder because I'm used to not having a checked in `.cargo/config.toml` which is why I wasn't sure what to do and made that PR. I guess that solution will work too.

About `rust-lld` being experimental. It is, but it's the first time I've ever had a compile issue with it enabled and I've had it for probably 2 years at this point. I know many people using the same config on windows. So I guess the better PR would be to document this in the windows build guide.

**Comment by osiewicz** [comment]
Yeah, documenting it in the build guide sounds like a good change, with a link to the issue and all that.

**Comment by IceSentry** [comment]
Here's the updated doc PR https://github.com/zed-industries/zed/pull/12859
